src/Mantasflowers.Services/Services/Payment/PaymentService.cs

```diff
@@ -0,0 +1,170 @@
+
+        public async Task<PostCreateCheckoutSessionResponse> CreateCheckoutSessionAsync(PostCreateCheckoutSessionRequest request)
+        {
+            var executionStrategy = _orderService.CreateExecutionStrategy();
+
+            /**
+             * As a unit of repetition
+             * Read more on: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/implement-resilient-entity-framework-core-sql-connections
+             */
+            var response = await executionStrategy.ExecuteAsync(() => CreateCheckoutSessionBodyAsync(request));
+
+            return response;
+        }
+
+        private async Task<PostCreateCheckoutSessionResponse> CreateCheckoutSessionBodyAsync(PostCreateCheckoutSessionRequest request)
+        {
+            using var transaction = await _orderService.BeginTransactionAsync();
+            var session = new Session();
+
+            try
+            {
+                var order = await _orderService.CreateOrderAsync(request.Order);
+                order = await _orderService.GetDetailedOrderAsync(order.Id); // Is there a better solution?
+
+                var lineItems = new List<SessionLineItemOptions>();
+
+                foreach (var orderItem in order.OrderItems)
+                {
+                    lineItems.Add(
+                        new SessionLineItemOptions
+                        {
+                            PriceData = new SessionLineItemPriceDataOptions
+                            {
+                                UnitAmountDecimal = decimal.Round(
+                                    orderItem.UnitPrice, 2,
+                                    MidpointRounding.AwayFromZero
+                                    ) * 100, // why stripe
+                                Currency = "eur",
+                                ProductData = new SessionLineItemPriceDataProductDataOptions
+                                {
+                                    Name = orderItem.Product.Name,
+                                    Description = orderItem.Product.ShortDescription,
+                                    Images = new List<string> { orderItem.Product.ThumbnailPictureUrl }
+                                }
+                            },
+                            Quantity = orderItem.Quantity
+                        });
+                }
+
+                var options = new SessionCreateOptions
+                {
+                    PaymentMethodTypes = new List<string> { "card" },
+                    Mode = "payment",
+                    LineItems = lineItems,
+                    SuccessUrl = request.SuccessUrl,
+                    CancelUrl = request.CancelUrl
+                };
+
+                session = await _stripeSessionService.CreateAsync(options);
+
+                await transaction.CommitAsync();
+            }
+            catch (StripeException)
+            {
+                await transaction.RollbackAsync();
```

Comment:
While this isn't something `UnitOfWork` would solve (or is designed to solve I think), I still don't really understand the point of the transaction here. Since the only transaction related thing is being done at the very top and if it fails, there's noting really to rollback. (I.e., `CreateOrderAsync` or  `GetDetailedOrderAsync` fails -> the entire method exists). Also dislike how `BeginTransactionAsync` and `CreateExecutionStrategy` are all over the service definitions now : (

Reply:
Repositories and Controllers are kinda out of question when it comes to transactions with multiple services involved, unless there is a better solution.